controllers/utils/dns.go

```diff
@@ -64,8 +64,30 @@ func Dig(fqdn string, edgeDNSServers ...DNSServer) (ips []string, err error) {
 	if err != nil {
 		return nil, fmt.Errorf("dig error: %s", err)
 	}
+	aRecords := make([]*dns.A, 0)
+	cnameRecords := make([]*dns.CNAME, 0)
 	for _, a := range ack.Answer {
-		ips = append(ips, a.(*dns.A).A.String())
+		switch v := a.(type) {
+		case *dns.A:
+			ips = append(ips, v.A.String())
+			aRecords = append(aRecords, v)
+		case *dns.CNAME:
+			cnameRecords = append(cnameRecords, v)
+		}
+	outer:
+		for _, cname := range cnameRecords {
+			for _, a := range aRecords {
+				if cname.Target == a.A.String() {
+					continue outer
+				}
+			}
+			// We do not have an A record for this CNAME
+			cnameIPs, err := Dig(cname.Target, edgeDNSServers...)
```

Comment:
We may hit the problem of infinite recursion. We can fix it by passing the max recursion as an argument:
```
func Dig(fqdn string, maxRecursion int, edgeDNSServers ...DNSServer) (ips []string, err error)
  if maxRecursion == 0 {
    return []string{}, nil
  }
```
There is no limit as per any RFC, so you can set it to any value you find reasonable. 8/16/32 seems to be common values

Reply:
Sounds like a good plan.  Should it be configurable, or just a referenced constant?  What should we do if we hit the maximum?  Return the current set?  Return an error?